tests/system/small/test_pandas.py

```diff
@@ -45,6 +45,56 @@ def test_concat_series(scalars_dfs):
     pd.testing.assert_series_equal(bf_result, pd_result)
 
 
+@pytest.mark.parametrize(
+    ("args"),
+    [
+        [["prefix1", "prefix2"], "_", None, ["bool_col", "int64_col"], False],
+        ["prefix", ["_", ","], False, ["int64_too", "string_col"], False],
+        [None, ".", True, ["time_col", "float64_col"], True],
+    ],
+)
+def test_get_dummies_dataframe(scalars_dfs, args):
+    scalars_df, scalars_pandas_df = scalars_dfs
+    prefix, prefix_sep, dummy_na, columns, drop_first = args
+
+    bf_result = bpd.get_dummies(
+        scalars_df,
+        prefix=prefix,
+        prefix_sep=prefix_sep,
+        dummy_na=dummy_na,
+        columns=columns,
+        drop_first=drop_first,
+    )
+    pd_result = pd.get_dummies(
+        scalars_pandas_df,
+        prefix=prefix,
+        prefix_sep=prefix_sep,
+        dummy_na=dummy_na,
+        columns=columns,
+        drop_first=drop_first,
+    )
+
+    pd.testing.assert_frame_equal(
+        bf_result.to_pandas(), pd_result, check_dtype=False
+    )  # dtype differences are expected
```

Comment:
I prefer the approach where we modify the pd_result as per the expected type, and then assert more strongly e.g. https://github.com/googleapis/python-bigquery-dataframes/blob/8e4451841ba09099b0ed5433f9102511741dfbed/tests/system/small/test_dataframe.py#L434

Reply:
Sounds good, I added a loop to change the mismatched datatypes.